src/ut_kvp.c

```diff
@@ -0,0 +1,487 @@
+        assert(true);
+        return 0;
+    }
+
+    //UT_LOG_DEBUG("Converted value: %llu", u64Value);
+    return u64Value;
+}
+
+ut_kvp_status_t ut_kvp_getStringField( ut_kvp_instance_t *pInstance, const char *pszKey, char *pszReturnedString, uint32_t uStringSize )
+{
+    struct fy_node *node = NULL;
+    struct fy_node *root = NULL;
+    const char *pString = NULL;
+
+    ut_kvp_instance_internal_t *pInternal = validateInstance(pInstance);
+
+    if (pInternal == NULL)
+    {
+        assert(pInternal != NULL);
+        return UT_KVP_STATUS_INVALID_INSTANCE;
+    }
+
+    if (pszKey == NULL)
+    {
+        assert(pszKey != NULL);
+        UT_LOG_ERROR("Invalid Param - pszKey");
+        return UT_KVP_STATUS_NULL_PARAM;
+    }
+
+    if ( pszReturnedString == NULL )
+    {
+        assert(pszReturnedString != NULL);
+        UT_LOG_ERROR("Invalid Param - pszReturnedString");
+        return UT_KVP_STATUS_NULL_PARAM;
+    }
+
+    /* Make sure we populate the returned string with zt before any other action */
+    *pszReturnedString=0;
+    if ( pInternal->fy_handle == NULL )
+    {
+        assert(pInternal->fy_handle != NULL);
+        UT_LOG_ERROR("No Data File open");
+        return UT_KVP_STATUS_NO_DATA;
+    }
+    // Get the root node
+    root = fy_document_root(pInternal->fy_handle);
+    if ( root == NULL )
+    {
+        assert( root != NULL );
+        UT_LOG_ERROR("Empty document");
+        return UT_KVP_STATUS_PARSING_ERROR;
+    }
+
+    // Find the node corresponding to the key
+    node = fy_node_by_path(root, pszKey, -1, FYNWF_DONT_FOLLOW);
+    if ( node == NULL )
+    {
+        //assert( node != NULL );
+        UT_LOG_ERROR("node not found: UT_KVP_STATUS_KEY_NOT_FOUND");
+        return UT_KVP_STATUS_KEY_NOT_FOUND;
+    }
+
+    if (node && fy_node_is_scalar(node))
+    {
+        // Get the string value
+        pString = fy_node_get_scalar0(node);
+        if ( pString == NULL )
+        {
+            //assert( pString != NULL );
+            UT_LOG_ERROR("field not found: UT_KVP_STATUS_KEY_NOT_FOUND");
+            return UT_KVP_STATUS_KEY_NOT_FOUND;
+        }
+    }
+    strncpy( pszReturnedString, pString, uStringSize );
+    return UT_KVP_STATUS_SUCCESS;
+}
+
+/** Static Functions */
+static ut_kvp_instance_internal_t *validateInstance(ut_kvp_instance_t *pInstance)
+{
+    ut_kvp_instance_internal_t *pInternal = (ut_kvp_instance_internal_t *)pInstance;
+
+    if ( pInstance == NULL )
+    {
+        assert(pInternal == NULL);
+        UT_LOG_ERROR("Invalid Handle");
+        return NULL;
+    }
+
+    if (pInternal->magic != UT_KVP_MAGIC)
+    {
+        assert(pInternal->magic != UT_KVP_MAGIC);
+        UT_LOG_ERROR("Invalid Handle - magic failure");
+        return NULL;
+    }
+
+    return pInternal;
+}
+
+static bool str_to_bool(const char *string)
+{
+    if ( strcasecmp( string, "true" ) == 0 )
+    {
+        return true;
+    }
+    if ( strcasecmp( string, "false" ) == 0 )
```

Comment:
you don't need to check for false, it's it's not == `true` then it must be `false`